compiler/z/codegen/SystemLinkagezOS.cpp

```diff
@@ -636,6 +663,180 @@ TR::S390zOSSystemLinkage::genCallNOPAndDescriptor(TR::Instruction* cursor, TR::N
    return cursor;
    }
 
+uint32_t
+TR::S390zOSSystemLinkage::generateCallDescriptorValue(TR::Node* callNode)
+   {
+   uint32_t result = 0;
+
+   if (cg()->comp()->target().is32Bit())
+      {
+      uint32_t returnValueAdjust = 0;
+
+      // 5 bit values for Return Value Adjust field of XPLLINK descriptor
+      enum ReturnValueAdjust
+         {
+         XPLINK_RVA_RETURN_VOID_OR_UNUSED  = 0x00,
+         XPLINK_RVA_RETURN_INT32_OR_LESS   = 0x01,
+         XPLINK_RVA_RETURN_INT64           = 0x02,
+         XPLINK_RVA_RETURN_FAR_POINTER     = 0x04,
+         XPLINK_RVA_RETURN_FLOAT4          = 0x08,
+         XPLINK_RVA_RETURN_FLOAT8          = 0x09,
+         XPLINK_RVA_RETURN_FLOAT16         = 0x0A,
+         XPLINK_RVA_RETURN_COMPLEX4        = 0x0C,
+         XPLINK_RVA_RETURN_COMPLEX8        = 0x0D,
+         XPLINK_RVA_RETURN_COMPLEX16       = 0x0E,
+         XPLINK_RVA_RETURN_AGGREGATE       = 0x10,
+         };
+
+      TR::DataType dataType = callNode->getDataType();
+
+      switch (dataType)
+         {
+         case TR::NoType:
+            returnValueAdjust = XPLINK_RVA_RETURN_VOID_OR_UNUSED;
+            break;
+
+         case TR::Int8:
+         case TR::Int16:
+         case TR::Int32:
+         case TR::Address:
+            returnValueAdjust = XPLINK_RVA_RETURN_INT32_OR_LESS;
+            break;
+
+         case TR::Int64:
+            returnValueAdjust = XPLINK_RVA_RETURN_INT64;
+            break;
+
+         case TR::Float:
+            returnValueAdjust = XPLINK_RVA_RETURN_FLOAT4;
+            break;
+
+         case TR::Double:
+            returnValueAdjust = XPLINK_RVA_RETURN_FLOAT8;
+            break;
+
+         default:
+            TR_ASSERT_FATAL(false, "Unknown datatype (%s) for call node (%p)", dataType.toString(), callNode);
+            break;
+         }
+
+      result |= returnValueAdjust << 24;
+
+      //
+      // Float parameter description fields
+      // Bits 8-31 inclusive
+      //
+
+      uint32_t parmAreaOffset = 0;
+
+#ifdef J9_PROJECT_SPECIFIC
+      TR::MethodSymbol* callSymbol = callNode->getSymbol()->castToMethodSymbol();
+      if (callSymbol->isJNI() && callNode->isPreparedForDirectJNI())
+         {
+         TR::ResolvedMethodSymbol * cs = callSymbol->castToResolvedMethodSymbol();
+         TR_ResolvedMethod * resolvedMethod = cs->getResolvedMethod();
+         // JNI Calls include a JNIEnv* pointer that is not included in list of children nodes.
+         // For FastJNI, certain calls do not require us to pass the JNIEnv.
+         if (!cg()->fej9()->jniDoNotPassThread(resolvedMethod))
+            parmAreaOffset += sizeof(uintptrj_t);
+
+         // For FastJNI, certain calls do not have to pass in receiver object.
+         if (cg()->fej9()->jniDoNotPassReceiver(resolvedMethod))
+            parmAreaOffset -= sizeof(uintptrj_t);
+         }
+#endif
+
+      uint32_t parmDescriptorFields = 0;
+
+
+      // WCode only logic follows for float parameter description fields
```

Comment:
Does this comment imply the following code is no longer relevant?  i.e., is it WCode-specific?

If still relevant, perhaps the comment should be removed.  I imagine the "WCode" now means "system linkage".

Reply:
It is not relevant. I copied the code over from previous code. It is not "WCode" related. I've removed the line. Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).